src/gateway/delegate-routing.ts

```diff
@@ -31,34 +38,73 @@ const onlines: { [index: string]: RoutingData } = {};
 router.use(
   "/findpeer",
   mixGetPost,
+  rateLimit({
+    enabled: gatewayConfigs.delegates.rateLimit.findPeerEnabled,
+    points: gatewayConfigs.delegates.rateLimit.findPeerLimit,
+    duration: gatewayConfigs.delegates.rateLimit.findPeerDuration,
+  }),
   asyncHandler(async (req, res, next) => {
     // @ts-ignore
-    let { id } = req.mixed;
+    const {id, timestamp, signature, requesterId} = req.mixed;
 
-    if (!id) throw `Missing parameter 'id'`;
+    if (!id || !timestamp || !signature || !requesterId)
+      throw `Missing parameters`;
 
-    const peerInfos: MuonNodeInfo[] = await NetworkIpc.filterNodes({
-      list: [id],
+    validateTimestamp(timestamp, findPeerValidPeriod);
+    const nodesInfo: MuonNodeInfo[] = await NetworkIpc.filterNodes({
+      list: [requesterId, id],
     })!;
+    const requesterPeerInfo = nodesInfo.find(peerInfo => peerInfo.peerId == requesterId);
+    const targetPeerInfo = nodesInfo.find(peerInfo => peerInfo.peerId == id);
 
-    if (peerInfos.length < 1) throw `PeerId '${id}' not found`;
 
-    const peerInfo = peerInfos[0];
+    if (!requesterPeerInfo)
+      throw `Invalid request source`;
+    if (!targetPeerInfo)
+      throw `PeerId '${id}' not found`;
 
-    const configs = loadGlobalConfigs('net.conf.json', 'default.net.conf.json');
-    const delegateRoutingTTL = parseInt(configs.routing.delegateRoutingTTL);
-    const onlinePeer = onlines[peerInfo.id];
-    if (onlinePeer && Date.now() - onlinePeer.timestamp > delegateRoutingTTL)
+    const targetOnlinePeer = onlines[targetPeerInfo.id];
+    if (!targetPeerInfo)
```

Comment:
This condition should be checked first. If peerInfo is not available on `onlines`, there is no need to check other things and can throw `not found` error.